middleware/gzip/filter.go

```diff
@@ -72,7 +84,7 @@ func DefaultMIMEFilter() MIMEFilter {
 // matches any of the registered ones. It returns true if
 // found and false otherwise.
 func (m MIMEFilter) ShouldCompress(r *http.Request) bool {
-	return m.Types.Contains(r.Header.Get("Content-Type"))
+	return m.Types.Contains(r.Header.Get("Accept"))
```

Comment:
I think we can get rid of it and move on for now until we find a proper way to handle it.